src/cargo/core/resolver/mod.rs

```diff
@@ -1108,22 +1096,16 @@ fn activate_deps_loop(
                         }
                     }
                     if !has_another && has_past_conflicting_dep && !backtracked {
-                        // we have not activated ANY candidates and
-                        // we are out of choices so add it to the cache
-                        // so our parent will know that we don't work
-                        let past = past_conflicting_activations
-                            .entry(dep.clone())
-                            .or_insert_with(Vec::new);
-                        if !past.contains(&conflicting_activations) {
-                            trace!(
-                                "{}[{}]>{} adding a meta-skip {:?}",
-                                parent.name(),
-                                cur,
-                                dep.name(),
-                                conflicting_activations
-                            );
-                            past.push(conflicting_activations.clone());
-                        }
+                        // we have not activated ANY candidates.
```

Comment:
Hm ok so if it's ok with you I'm gonna zero in on this block and see if I can understand it. The comment here mentions that we haven't activated any candidates, but we've activated `candidate` above, right? We're concluding here, however, that this activation is doomed to fail, which leads us to filter out the backtrack stack.

How come we conclude here, after activating, to filter the backtrack stack? I'd naively expect this logic to be above when we fail to activate a frame. Basically my thinking would be that we, above, conclude that conflicting activations make this dependency fail to activate. That naturally means that any backtrack frame which has all our conflicts activated is also doomed to fail.

In light of that, do we still need the `!has_another` condition and the `!backtracked` condition here? I'm not 100% sure what those are guarding against, but I'm sure I'm missing something!

Reply:
I think you are understanding this as well as I am. I.E. I am confused as well.

"but we've activated candidate above, right?" Yes.
"expect this logic to be above when we fail to activate a frame" We don't need it up above because `find_candidate` basically does it for us.
"In light of that, do we still need" any of this block? I don't know why we do. I pushed a commit removing it, time to figure out a better fix for the new test.

Reply:
Oh interesting! So removing this `retain` meant the test case added here still executed quickly? I see what you mean about `find_candidate` basically doing this already yeah